reana_workflow_engine_serial/tasks.py

```diff
@@ -40,9 +41,12 @@ def escape_shell_arg(shell_arg):
 
     return "%s" % shell_arg.replace('"', '\\"')
 
+
 def load_json(ctx, param, value):
     """Callback function for click option"""
-    return json.loads(value)
+    value = str.encode(value[1:])
+    return json.loads(base64.standard_b64decode(value).decode())
```

Comment:
Hmm, why not simply `json.loads(base64.standard_b64decode(value))` without the `value[1:]` bit and the `decode()` bit?

Reply:
I am not sure if I understood you correctly.

Workflow json is passed to cmd builder as bytes and then converted to string, so that's why there is `b` at the begining . 

`--workflow-json 'b'eyJzdGVwcyI6IFt7ImNvbW1hbmRzIjogWyJweXRob24gXCIke2hlbGxvd29ybGR9XCIgLS1zbGVlcHRpbWUgJHtzbGVlcHRpbWV9IC0taW5wdXRmaWxlIFwiJHtpbnB1dGZpbGV9XCIgLS1vdXRwdXRmaWxlIFwiJHtvdXRwdXRmaWxlfVwiIl0sICJlbnZpcm9ubWVudCI6ICJweXRob246Mi43In1dfQ==''`

If I will try to decode with first character, it won't work. 

```
value
'beyJzdGVwcyI6IFt7ImNvbW1hbmRzIjogWyJweXRob24gXCIke2hlbGxvd29ybGR9XCIgLS1zbGVlcHRpbWUgJHtzbGVlcHRpbWV9IC0taW5wdXRmaWxlIFwiJHtpbn

json.loads(base64.standard_b64decode(value))
UnicodeDecodeError: UnicodeDecodeError('utf-8', b'm\xec\x89\xcd\xd1\x95\xc1\xcc\x88\xe8\x81m\xec\x89\x8d\xbd\xb5\xb5\x85\xb9\x91\xcc\x88\xe8\x81l\x89\xc1\xe5\xd1\xa1\xbd\xb8\x8, 1, 3, 'invalid continuation byte')
```

```
json.loads(base64.standard_b64decode(value[1:]))
{'steps': [{'commands': ['python "${helloworld}" --sleeptime ${sleeptime} --inputfile "${inputfile}" --outputfile "${outputfile}"'], 'environment': 'python:2.7'}]}
```

Definitely, it is not a clean solution and needs to be improved.

Reply:
@roksys Ah, OK, if you keep leading `b` then yes...